src/wp-admin/includes/image-edit.php

```diff
@@ -785,6 +785,9 @@ function wp_restore_image( $post_id ) {
 		}
 	}
 
+	// Perform operations to save the sources properties, specifically for the `full` size image due this is a virtual image size.
+	wp_uploads_restore_image( $post_id, $meta );
```

Comment:
We need to integrate the relevant logic from the plugin directly with the core logic, rather than simply adding the plugin functions to core and calling them. The way this is implemented in the plugin is quite hacky because WordPress core did not provide sufficient integration points (actions/filters) to implement it in a cleaner way. But now that we're actually working in core, we should integrate our changes directly with `wp_save_image()` and `wp_restore_image()` logic.